linera-core/src/client/mod.rs

```diff
@@ -871,6 +871,9 @@ where
         #[cfg(with_metrics)]
         let _latency = metrics::PREPARE_CHAIN_LATENCY.measure_latency();
 
+        let mutex = self.state().client_mutex();
+        let _guard = mutex.lock_owned().await;
```

Comment:
Do we need two separate lines when `lock_owned` is used?

Reply:
We actually do! `self.state()` returns a `ChainGuard` that can't be held across an await point… specifically so we wouldn't ever lock an entry in the chain states map locked.

And it looks like Rust would drop that guard only after the whole _expression_, i.e. after the `await`.